diff/diff_test.go

```diff
@@ -818,6 +819,40 @@ func TestParseMultiFileDiffAndPrintMultiFileDiff(t *testing.T) {
 	}
 }
 
+func TestParseMultiFileDiffAndPrintMultiFileDiffIncludingTrailingContent(t *testing.T) {
+	diffData, err := ioutil.ReadFile(filepath.Join("testdata", "sample_multi_file_minuses_pluses_trailing_content.diff"))
```

Comment:
Does the fact that there are minuses and pluses in the diff play a role? If not, how about creating a separate diff in the testdata that is specific to this test and has content before, between and after the diffs? That way the test case would make clear where each bit of additional non-diff content lands.

Reply:
From reviewing the code in `ReadExtendedHeaders()`, it appears as though all lines are hoovered up into the extended headers array until we hit something that doesn't look like an extended header, e.g. a `---` or a `diff ...`.

I am not familiar with the formalities of the diff format tbh. This library's code seems to imply that it's impossible to tell the difference between leading non-diff content, and extended headers. Trailing content was the exception, because there was no diff to place the extended headers into.

I've changed the test to demonstrate this, asserting on the whole content of the diff, which has leading/trailing content. Please let me know if I've missed the point!

Reply:
> I've changed the test to demonstrate this, asserting on the whole content of the diff, which has leading/trailing content. Please let me know if I've missed the point!

You didn't - test does exactly what I was looking for, thanks!